src/Nox.Generator/Domain/Entity.template.cs

```diff
@@ -251,17 +251,92 @@ public virtual void ClearDomainEvents()
         {{- end }}
     }
 {{- end }}
-{{- for relationship in entity.OwnedRelationships #TODO how to reuse as partial template?}}
+
+{{- ######################################### Owned Relationships ###################################################### -}}
+
+{{- for relationship in entity.OwnedRelationships }}
 
     /// <summary>
     /// {{entity.Name}} {{relationship.Description}} {{relationship.Relationship}} {{relationship.EntityPlural}}
     /// </summary>
     {{- if relationship.Relationship == "ZeroOrMany" || relationship.Relationship == "OneOrMany"}}
-    public virtual List<{{relationship.Entity}}> {{relationship.Name}} { get; set; } = new();
-    {{- else}}
-     public virtual {{relationship.Entity}}{{if relationship.Relationship == "ZeroOrOne"}}?{{end}} {{relationship.Name}} { get; set; } = null!;
-    {{-end}}
-{{- end }}
+    public virtual List<{{relationship.Entity}}> {{relationship.Name}} { get; private set; } = new();
+	{{- else}}
+    public virtual {{relationship.Entity}}{{if relationship.Relationship == "ZeroOrOne"}}?{{end}} {{relationship.Name}} { get; private set; }{{if relationship.Relationship == "ExactlyOne"}} = null!;{{end}}
+    {{- if relationship.ShouldGenerateForeignOnThisSide && (relationship.Related.Entity.Keys | array.size) > 0 }}
```

Comment:
this is used in normal relations, where we need to define where the relation is going to be generated.

owned entities, the relation is on the main entity always, and are fully manage in the main entity, Id is not needed. Do we need this change in the scope of this ticket?

Reply:
Thanks @rochar, let's change it with this ticket.